kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -232,4 +235,109 @@ default boolean isFinalizerValid(String finalizer) {
   default boolean removeFinalizer(String finalizer) {
     return getMetadata().getFinalizers().remove(finalizer);
   }
+
+  default boolean hasOwnerReferenceFor(HasMetadata owner) {
+    return getOwnerReferenceFor(owner).isPresent();
+  }
+
+  default boolean hasOwnerReferenceFor(String ownerUid) {
+    return getOwnerReferenceFor(ownerUid).isPresent();
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(HasMetadata owner) {
+    if (owner == null) {
+      return Optional.empty();
+    }
+
+    final String ownerUID = owner.getMetadata().getUid();
+    return getOwnerReferenceFor(ownerUID);
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(String ownerUid) {
+    if (ownerUid == null || ownerUid.isEmpty()) {
+      return Optional.empty();
+    }
+
+    return getMetadata().getOwnerReferences().stream()
+      .filter(o -> ownerUid.equals(o.getUid()))
+      .findFirst();
+  }
+
+  default OwnerReference addOwnerReference(HasMetadata owner) {
```

Comment:
With a quick glance, it looks as if the purpose of this code is validation. 
However, these are validations that won't be performed when the user is creating objects using the builders.

So, I think that a more suitable place for these validations is in the constructor of `OwnerReference` and `ObjectMeta`.
And yes, I think that OwnerRefernce is generated, so way may consider managing it ourselves.

Reply:
Sure, however, the same could have been said of the finalizer support that got added a while ago… so is your opinion that we should that code to `ObjectMeta` as well?

Reply:
Another issue with moving the validation to `OwnerReference`'s constructor is the fact that there needs to be a no-arg constructor for serialization meaning that we cannot ensure that the "appropriate" constructor will be called. 
I agree that design-wise it would make more sense to move the code elsewhere, however that would mean managing manually two additional classes: `ObjectMeta` and `OwnerReference`.

Reply:
Regarding the no-arg constructor, there is nothing to validate there, given it's `no-arg` so that doesn't change anything.
Regarding the finalizer, I missed that change,  so I can't really comment on that.
Regarding, managing two classes, I say it's a fair trade.

Reply:
I would suggest we (you?) do that change in a different PR.